plugins/out_nats/nats.c

```
@@ -159,69 +206,161 @@ static void cb_nats_flush(struct flb_event_chunk *event_chunk,
+     * The handshake state is tracked in the connection's own user_data
+     * field (unused for upstream connections): connections are zeroed on
+     * creation, so a fresh connection always runs the handshake, while a
+     * recycled one keeps the mark and skips it. A connection that fails an
+     * I/O operation is destroyed by the upstream layer rather than
+     * recycled, and the keepalive queue destroys connections dropped by
+     * the server, so no stale mark can ever skip the handshake on a new
+     * connection.
+     */
+    if (u_conn->user_data == NULL) {
+        ret = flb_io_net_write(u_conn,
+                               NATS_CONNECT,
```

Cppcheck (2.21.0)

[error] There is an unknown macro here somewhere. Configuration is required. If FLB_VERSION_STR is a macro then please configure it.

(unknownMacro)